content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,668 @@
+---
+title: "Peace of Mind with Cloud Secret Providers"
+
+date: 2020-04-29T14:26:35-07:00
+draft: true
+meta_desc: "Learn how to encrypt your Pulumi secrets with a cloud secrets provider to ensure secrets never leave your control"
+meta_image: meta.png
+authors:
+    - lee-briggs
+tags:
+    - features
+    - security
+---
+
+The secrets in your infrastructure are a vital part of your security model, and provisioning infrastructure is an inherently privileged process. Back in [Pulumi 0.17.12](https://www.pulumi.com/blog/managing-secrets-with-pulumi/) we introduced client-side secret encryption and started encrypting secret configuration values inside the Pulumi state so that users could be confident their passwords, tokens and other secret values were viewable only by them while managing their infrastructure.
+Our first iteration of the client-side encryption used a passphrase for encrypting the secret, however, remembering that passphrase and sharing it securely with team members is less than ideal.
+That's why in our 0.17.28 release, we added support for "Cloud Secret Providers", giving users fill confidence that their sensitive values are for their eyes only.
+
+<!--more-->
+
+Pulumi supports encryption via [AWS KMS](https://aws.amazon.com/kms/), [Azure KeyVault](https://azure.microsoft.com/en-us/services/key-vault/), [Google Cloud KMS](https://cloud.google.com/kms) and [Hashicorp Vault](https://www.vaultproject.io/). This post shows you how to create an AWS KMS in a stack using Pulumi, and then use the key to encrypt values in subsequent stacks.
```

Comment:
Do we need to suggest that this is a series of posts? Or can we just say that we will use AWS as an example of how to do this?

Reply:
I wasn't planning on doing this for Azure and GCP, certainly not in the short term. I'll reword it